src/braket/tasks/local_quantum_task.py

```diff
@@ -30,38 +44,306 @@ class LocalQuantumTask(QuantumTask):
+            program = circuit.to_ir(ir_type=IRType.JAQCD)
+            qubits = circuit.qubit_count
+            return [program, qubits, shots] + list(self._args)
+        raise NotImplementedError(f"{type(simulator)} does not support qubit gate-based programs")
+
+    @_convert_to_sim_format.register
+    def _(
+        self,
+        problem: Problem,
+        shots: Optional[int] = None,
+        inputs: Optional[Dict[str, float]] = None,
+    ) -> list:
+        simulator = self._delegate
+        if DeviceActionType.ANNEALING not in simulator.properties.action:
+            raise NotImplementedError(
+                f"{type(simulator)} does not support quantum annealing problems"
+            )
+        ir = problem.to_ir()
+        return [ir, shots] + list(self._args)
+
+    @_convert_to_sim_format.register
+    def _(
+        self,
+        program: Program,
+        shots: Optional[int] = None,
+        inputs: Optional[Dict[str, float]] = None,
+    ) -> list:
+        simulator = self._delegate
+        if DeviceActionType.OPENQASM not in simulator.properties.action:
+            raise NotImplementedError(f"{type(simulator)} does not support OpenQASM programs")
+        if inputs:
+            inputs_copy = program.inputs.copy() if program.inputs is not None else {}
+            inputs_copy.update(inputs)
+            program = Program(
+                source=program.source,
+                inputs=inputs_copy,
+            )
+        return [program, shots] + list(self._args)
+
+    @_convert_to_sim_format.register
+    def _(
+        self,
+        program: AnalogHamiltonianSimulation,
+        shots: Optional[int] = None,
+        inputs: Optional[Dict[str, float]] = None,
+    ) -> list:
+        simulator = self._delegate
+        if DeviceActionType.AHS not in simulator.properties.action:
+            raise NotImplementedError(
+                f"{type(simulator)} does not support analog Hamiltonian simulation programs"
+            )
+        return [program.to_ir(), shots] + list(self._args)
+
+    @_convert_to_sim_format.register
+    def _(
+        self,
+        program: AHSProgram,
+        shots: Optional[int] = None,
+        inputs: Optional[Dict[str, float]] = None,
+    ) -> list:
+        simulator = self._delegate
+        if DeviceActionType.AHS not in simulator.properties.action:
+            raise NotImplementedError(
+                f"{type(simulator)} does not support analog Hamiltonian simulation programs"
+            )
+        return [program, shots] + list(self._args)
+
+    @singledispatchmethod
+    def _run_internal(
+        self,
+        task_specification: Union[
+            Circuit, Problem, Program, AnalogHamiltonianSimulation, AHSProgram
+        ],
+        shots: Optional[int] = None,
+        *args,
+        **kwargs,
+    ) -> Union[GateModelQuantumTaskResult, AnnealingQuantumTaskResult]:
+        raise NotImplementedError(f"Unsupported task type {type(task_specification)}")
```

Comment:
How are you testing this? Is there any real code path that passes _convert_to_sim_format but fails here?